x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MachineLearningFeatureSetUsage.java

```diff
@@ -24,28 +25,37 @@
     public static final String DETECTORS = "detectors";
     public static final String FORECASTS = "forecasts";
     public static final String MODEL_SIZE = "model_size";
+    public static final String NODE_COUNT = "node_count";
 
     private final Map<String, Object> jobsUsage;
     private final Map<String, Object> datafeedsUsage;
+    private int nodeCount;
 
     public MachineLearningFeatureSetUsage(boolean available, boolean enabled, Map<String, Object> jobsUsage,
-                                          Map<String, Object> datafeedsUsage) {
+                                          Map<String, Object> datafeedsUsage, int nodeCount) {
         super(XPackField.MACHINE_LEARNING, available, enabled);
         this.jobsUsage = Objects.requireNonNull(jobsUsage);
         this.datafeedsUsage = Objects.requireNonNull(datafeedsUsage);
+        this.nodeCount = nodeCount;
     }
 
     public MachineLearningFeatureSetUsage(StreamInput in) throws IOException {
         super(in);
         this.jobsUsage = in.readMap();
         this.datafeedsUsage = in.readMap();
+        if (in.getVersion().onOrAfter(Version.V_6_5_0)) {
+            this.nodeCount = in.readInt();
+        }
```

Comment:
Rather than use `0` in the case of "unknown" in a mixed version cluster it would be nicer to have a specific "unknown" value, say `-1`, and then not include the count in the JSON output if it's unknown.

For example:

```
        if (in.getVersion().onOrAfter(Version.V_6_5_0)) {
            this.nodeCount = in.readInt();
        } else {
            this.nodeCount = -1;
        }
```